tests/test_openeo_gfmap/test_utils.py

```diff
@@ -54,6 +58,29 @@ def test_query_cdse_catalogue():
     assert decision == "DESCENDING"
 
 
+def test_query_cdse_catalogue_with_s1_gap():
```

Comment:
This test is currently an integration tests because you actively query on the backend. The test will not only fail in case the code changes, but also in case something changes on the back-end or if there are back-end issues.

Best practice would be to mock the output of areas = s1_area_per_orbitstate_vvvh(backend, spatial_extent, temporal_extent) for all cases in the selection criteria and check if the right decision is made.

Reply:
I took as example the test before that one which does exactly the same. So if this separation is required, it needs to be done on a broader scale. This will be out of scope for me personally.

Reply:
but in a future PR we do plan to migrate the current integration tests to a cron job and only keep the true unit tests. The more we already have in place of those, the better.

Reply:
@HansVRP To transform this in unit-tests, what do you think if I save the results of the cdse query in a test resource file, and then create a mock function in the unit tests `mock_cdse_query(collection, bounds, temporal_extent, **additional_parameters)` ?

The mock function would compute the hash of the input parameters (on which there should be only a few variation in the unit tests) and then fetches the results like so: `test_openeo_gfmap/resources/<first_digits_of_hash>_cdse_query.json`

Reply:
Well I think any mock is good if you do not actually connect to the cdse backend. 

I have included in a previous PR a few examples where I mock around functions, in order to only test the functionality you are interested in. Maybe you can draw some inspiration there? 

But as long as you do not directly connect to cdse, I think its good!

Reply:
Added the mockup